### PR TITLE
fix(integration-tests): fix false negatives in SSR tests

### PR DESCRIPTION
**Pull request by vladar**
Turns out our `ssr` integration tests were not working correctly. They were not capturing errors because of the test setup. Basically, jest runs `test-output` script under the hood:

https://github.com/gatsbyjs/gatsby/blob/132d829e01b3b2ca50bfbe4533ae47b57b55a50c/integration-tests/ssr/__tests__/ssr.js#L27-L31

This script was returning code `0` but also emitted this _warning_ to the standard output:

```
(node:952) UnhandledPromiseRejectionWarning: Error: Cannot find module 'cheerio'
```

In other words, the script was failing but tests were passing. This PR handles promise rejections correctly, adds all missing dependencies, and adds some sanity checks to the test case to make sure we get the expected output.

### CODE REVIEW

**Review by pieh — APPROVED** [review]
Mystery solved :) Thank you!
